Gravity/Gravity/Base/BaseDto.cs

```diff
@@ -202,9 +160,11 @@ private static object ConvertPropertyValue(PropertyInfo property, RdoFieldType f
 
 				case RdoFieldType.SingleObject:
 					{
-						if ((int)propertyValue > 0)
+						int artifactId = (int)propertyValue.GetType().GetProperty("ArtifactId")?.GetValue(propertyValue, null);
```

Comment:
Cast object to BaseDto instead of using reflection (yes, this issue preceded you).